store/src/main/java/com/dropbox/android/external/store4/impl/RealStore.kt

```diff
@@ -56,15 +56,17 @@ internal class RealStore<Key, Input, Output>(
             }
     private val memCache = memoryPolicy?.let {
         CacheBuilder.newBuilder()
-                .also {
-                    if (memoryPolicy.hasAccessPolicy()) {
-                        it.expireAfterAccess(memoryPolicy.expireAfterAccess, memoryPolicy.expireAfterTimeUnit)
-                    }
-                    if (memoryPolicy.hasWritePolicy()) {
-                        it.expireAfterWrite(memoryPolicy.expireAfterWrite, memoryPolicy.expireAfterTimeUnit)
-                    }
-                    if (memoryPolicy.hasMaxSize()) {
-                        it.maximumSize(memoryPolicy.maxSize)
+                .apply {
+                    when {
```

Comment:
when is not equivalent to the previous code here. it will only execute the first matched branch. this means that if a policy has both access policy and max size for example, only the first will be applied.

Reply:
You are right, that was a mistake! Fixed it!